### DISCUSSION TITLE
Comments on ATTiRe and workflow for importing logs.

### DISCUSSION THREAD

**Discussion by improg00n** [SecurityRiskAdvisors/VECTR#174] [Ideas]
Hey, I was encouraged by Galen to start a discussion, so here goes:

Some comments on the workflow for importing data.
First, the logging format ATTiRe has some custom fields, which it would be nice to have documented.
It would also be nice with a bit more documentation on what types are expected for the ATTiRe format.
An issue I have run into has been time parsing, where it mentions being iso-8601 format, however, it does not actually support all cases, but as far as I can tell. only a very specific encoding of the iso-8601.

Another issue is the UI. Currently importing larger logs, the workflow consists of being able to ''select all' cases, but having to scroll to the bottom of the window to submit. 
Also making it more clear where to import logs would be desired, though it is an "learn-once" experience.
 The "Assesment actions" button, is very clear, but it also hides functionality.

And finally, I would like to hear your thoughts on providing a "status" field. Given the status field would say something like "succeded/failed", would it be possible to have vectr automatically check off "blocked" in this instance?

Best regards

**Comment by thebleucheese** [comment]
@improg00n Thanks for the feedback. I agree about status / execution success! 

We've been discussing adding something similar to ATTiRe. I want to think a bit on the level of care we take with marking success / failure - maybe a third option for did not finish or incomplete?

Maybe we need some way to capture the exit code too. There are a number of failure scenarios that can occur due to test logic or configuration and it complicates the logic we include in the executor to determine if execution failure is due to endpoint controls or otherwise. I don't know if we can assume block when tests don't run successfully.

I plan to get some UI workflow improvements in on the VECTR side as well.
